File: app/src/main/java/de/ba/railroadclient/ws/ConnectTask.java
package de.ba.railroadclient.ws;

import android.os.AsyncTask;
import android.util.Log;

import org.eclipse.jetty.websocket.client.WebSocketClient;

import java.net.URI;

import ws.MessageAdapter;

/**
 * AsyncTask for the WebSocket connection, used in {@link LocomotiveWebSocketClient#connect(String)}.
 */
@SuppressWarnings("ALL")
class ConnectTask extends AsyncTask<String, String, Void> {

    /**
     * Provides a means of establishing connections to remote websocket endpoints
     */
    private org.eclipse.jetty.websocket.client.WebSocketClient client;

    /**
     * JETTY basic Web Socket implementation
     */
    private MessageAdapter webSocket;

    /**
     * Stores the WebSocketClient and MessageAdapter
     * @param client provides a means of establishing connections to remote websocket endpoints
     * @param webSocket JETTY basic Web Socket implementation
     */
    ConnectTask(WebSocketClient client, MessageAdapter webSocket) {
        this.client = client;
        this.webSocket = webSocket;
    }

    /**
     * Connects the {@link #client} using the {@link #webSocket} and given URL.
     *
     * @param uris Array of URLs, first URL is used
     * @return null
     */
    @Override
    protected Void doInBackground(String... uris) {

        if (uris == null || uris.length == 0) {
            Log.d("ConnectTask", "can not connect to an empty URL");
            return null;
        }

        try {
            URI echoUri = new URI(uris[0]);
            client.connect(webSocket, echoUri);
            Log.d("ConnectTask", "Connecting to: " + echoUri);

        } catch (Throwable t) {
            Log.d("ConnectTask", "can not connect", t);
        }

        return null;
    }
}

File: app/src/main/java/de/ba/railroadclient/ws/ConnectTask.kt
package de.ba.railroadclient.ws

import org.eclipse.jetty.websocket.client.WebSocketClient
import ws.MessageAdapter
import android.os.AsyncTask
import android.util.Log
import ws.CraneWebSocketFacade
import de.ba.railroadclient.ws.ConnectTask
import ws.LocomotiveWebSocketFacade
import ws.SwitchGroupWebSocketFacade
import ws.BarrierGroupWebSocketFacade
import java.net.URI

/**
 * AsyncTask for the WebSocket connection, used in [LocomotiveWebSocketClient.connect].
 */
internal class ConnectTask
/**
 * Stores the WebSocketClient and MessageAdapter
 * @param client provides a means of establishing connections to remote websocket endpoints
 * @param webSocket JETTY basic Web Socket implementation
 */(
    /**
     * Provides a means of establishing connections to remote websocket endpoints
     */
    private val client: WebSocketClient,
    /**
     * JETTY basic Web Socket implementation
     */
    private val webSocket: MessageAdapter?
) : AsyncTask<String?, String?, Void?>() {

    /**
     * Connects the [.client] using the [.webSocket] and given URL.
     *
     * @param uris Array of URLs, first URL is used
     * @return null
     */
    @Deprecated("Deprecated in Java")
    override fun doInBackground(vararg params: String?): Void? {
        if (params == null || params.size == 0) {
            Log.d("ConnectTask", "can not connect to an empty URL")
            return null
        }
        try {
            val echoUri = URI(params[0])
            client.connect(webSocket, echoUri)
            Log.d("ConnectTask", "Connecting to: $echoUri")
        } catch (t: Throwable) {
            Log.d("ConnectTask", "can not connect", t)
        }
        return null
    }
}